pythia/scripts/eval.py

```diff
@@ -0,0 +1,129 @@
+        # Set up scorers
+        # =================================================
+        print("setting up scorers...")
+        scorers = [
+            (Bleu(4), ["Bleu_1", "Bleu_2", "Bleu_3", "Bleu_4"]),
+            (Meteor(), "METEOR"),
+            (Rouge(), "ROUGE_L"),
+            (Cider(), "CIDEr"),
+            (Spice(), "SPICE"),
+        ]
+
+        # =================================================
+        # Compute scores
+        # =================================================
+        for scorer, method in scorers:
+            print("computing %s score..." % (scorer.method()))
+            score, scores = scorer.compute_score(gts, res)
+            if type(method) == list:
+                for sc, scs, m in zip(score, scores, method):
+                    self.setEval(sc, m)
+                    self.setImgToEvalImgs(scs, gts.keys(), m)
+                    print("%s: %0.3f" % (m, sc))
+            else:
+                self.setEval(score, method)
+                self.setImgToEvalImgs(scores, gts.keys(), method)
+                print("%s: %0.3f" % (method, score))
+        self.setEvalImgs()
+
+    def setEval(self, score, method):
+        self.eval[method] = score
+
+    def setImgToEvalImgs(self, scores, imgIds, method):
+        for imgId, score in zip(imgIds, scores):
+            if imgId not in self.imgToEval:
+                self.imgToEval[imgId] = dict()
+                self.imgToEval[imgId]["image_id"] = imgId
+            self.imgToEval[imgId][method] = score
+
+    def setEvalImgs(self):
+        self.evalImgs = [eval for imgId, eval in self.imgToEval.items()]
+
+
+def calculate_metrics(imgIds, datasetGTS, datasetRES):
+    imgToAnnsGTS = {id: [] for id in imgIds}
+    for ann in datasetGTS["annotations"]:
+        imgToAnnsGTS[ann["image_id"]] += [ann]
+
+    imgToAnnsRES = {id: [] for id in imgIds}
+    for ann in datasetRES["annotations"]:
+        imgToAnnsRES[ann["image_id"]] += [ann]
+
+    evalObj = COCOEvalCap(imgIds, imgToAnnsGTS, imgToAnnsRES)
+    evalObj.evaluate()
+    return evalObj.eval
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Image captioning metrics")
+    parser.add_argument("--reference_json", help="Path to reference captions json")
+    parser.add_argument("--predicted_json", help="Path to predicted captions json")
+    args = parser.parse_args()
+
+    with open(args.reference_json, "r") as f:
+        captions = json.load(f)
+
+    references = []
+    img_ids = []
+
+    for img in captions["images"]:
+        if img["split"] == "test":
+            for c in img["sentences"]:
+                d = {}
+                d["image_id"] = c["imgid"]
+                img_ids.append(c["imgid"])
+                d["caption"] = c["raw"]
+                references.append(d)
+    img_ids = list(set(img_ids))
+
+    with open(args.predicted_json, "r") as f:
+        preds = json.load(f)
+
+    datasetGTS = {"annotations": references}
```

Comment:
Let's not use camelCase styling in our code. `datasets_gts`. For cocoeval it is fine as it is an external library but we shouldn't follow that style in our code.